app/components/tic-tac-toe/game/component.js

```diff
@@ -0,0 +1,149 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { cloneDeep, random } from 'lodash';
+
+export default class TicTacToeComponent extends Component {
+  @tracked blocks = Array.from({ length: 9 }, (_, id) => ({
+    symbol: null,
+    id,
+  }));
+  @tracked nextMove = '𝗫';
+  @tracked isStarted = false;
+  @tracked winner = null;
+  @tracked isSingleplayer = true;
+  @tracked computerPossibleBlocks = [];
+
+  matrix = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+
+  get shouldBeDisabled() {
+    return Boolean(this.isStarted);
+  }
+
+  get gameMode() {
+    if (this.isSingleplayer) {
+      return 'Singleplayer';
+    }
+    return 'Multiplayer';
+  }
+
+  get shouldBeAbleToClick() {
+    return Boolean(!this.winner);
+  }
+
+  get nobodyWin() {
+    return Boolean(
+      this.blocks.find((block) => {
+        return block.id !== null;
+      })
+    );
+  }
+
+  @action
+  onChangeGameMode() {
+    this.isSingleplayer = !this.isSingleplayer;
+  }
+
+  @action
+  onClick(event) {
+    if (this.isSingleplayer) {
+      if (this.shouldBeAbleToClick) {
+        let newGrid = this.blocks;
+        let blockState = newGrid.find(
+          (block) => block.id === JSON.parse(event.target.id)
+        );
```

Comment:
🤔 
```suggestion
        let blockState = newGrid.find(
          ({ id }) => id === JSON.parse(event.target.id)
        );
```

Also you can destructure `event` to `target` and on thing - consider to replace all `let`'s to `const` if you know you will not update variable

Reply:
ping pong 🏓